### PR TITLE
minor notes explaining rmw_qos_profile_t

### PR DESCRIPTION
**Pull request by mlautman**

### CODE REVIEW

**Comment by dirk-thomas** [comment]
If the helpful text should already in the API documentation it needs to be in a docblock.

File: rmw/include/rmw/types.h [hunk 1]
```diff
@@ -198,9 +198,11 @@ enum RMW_PUBLIC_TYPE rmw_qos_durability_policy_t
   RMW_QOS_POLICY_DURABILITY_VOLATILE
 };
 
+// rmw_qos_profile_t: ROS MiddleWare quality of service profile
```
Comment (mlautman) [inline rmw/include/rmw/types.h]:
```suggestion
/// ROS MiddleWare quality of service profile.
```

File: rmw/include/rmw/types.h [hunk 2]
```diff
@@ -198,9 +198,11 @@ enum RMW_PUBLIC_TYPE rmw_qos_durability_policy_t
   RMW_QOS_POLICY_DURABILITY_VOLATILE
 };
 
+/// ROS MiddleWare quality of service profile.
 typedef struct RMW_PUBLIC_TYPE rmw_qos_profile_t
 {
   enum rmw_qos_history_policy_t history;
+  // depth is the size of the message queue
```
Comment (mlautman) [inline rmw/include/rmw/types.h]:
```suggestion
  //! Size of the message queue 
```

File: rmw/include/rmw/types.h [hunk 3]
```diff
@@ -198,9 +198,11 @@ enum RMW_PUBLIC_TYPE rmw_qos_durability_policy_t
   RMW_QOS_POLICY_DURABILITY_VOLATILE
 };
 
+/// ROS MiddleWare quality of service profile.
 typedef struct RMW_PUBLIC_TYPE rmw_qos_profile_t
 {
   enum rmw_qos_history_policy_t history;
+  //! Size of the message queue
```
Comment (mlautman) [inline rmw/include/rmw/types.h]:
```suggestion
  //! Size of the message queue.
```

File: rmw/include/rmw/types.h [hunk 4]
```diff
@@ -198,9 +198,11 @@ enum RMW_PUBLIC_TYPE rmw_qos_durability_policy_t
   RMW_QOS_POLICY_DURABILITY_VOLATILE
 };
 
+/// ROS MiddleWare quality of service profile.
```
Comment (mlautman) [inline rmw/include/rmw/types.h]:
```suggestion
//! ROS MiddleWare quality of service profile.
```

**Comment by mlautman** [comment]
@dirk-thomas Updated for Doxygen

**Comment by mlautman** [comment]
@dirk-thomas Ping

**Pull request merged**